packages/playground/src/components/Editor.tsx

```diff
@@ -0,0 +1,52 @@
+import { memo, useCallback, useState } from 'react';
+import MonacoEditor from '@monaco-editor/react';
+
+const monacoEditorOptions = {
+  minimap: {
+    enabled: false,
+  },
+  automaticLayout: true,
+};
+
+const Editor: React.FC<{ title: string; code: string; onChange: (code: string) => void }> = memo(
+  ({ title, code: initialCode, onChange }) => {
+    const [valid, setValid] = useState(true);
+    const [code, setCode] = useState(initialCode);
+
+    const onCodeChange = useCallback((code: string | undefined) => {
+      if (!code) return;
+
+      try {
+        const parsedCode = JSON.parse(code);
+        setValid(true);
+        setCode(code);
+        onChange(parsedCode);
+      } catch (err) {
+        setValid(false);
+        setCode(code);
+      }
+    }, []);
```

Comment:
Oh oops, you'll want the set functions as dependencies too since they are props
```suggestion
    }, [setValid, setCode, onChange]);
```